hs_core/hydroshare/resource.py

```diff
@@ -566,15 +566,19 @@ def create_new_version_resource(ori_res, new_res, user):
                                         fed_resource_file_name_or_path=f.fed_resource_file_name_or_path,
                                         fed_resource_file_size=f.fed_resource_file_size)
         elif f.fed_resource_file:
-            ResourceFile.objects.create(content_object=new_res,
-                                        fed_resource_file=os.path.join('{zone}/{res_id}/data/contents/{file_name}'.format(
-                                            zone=ori_res.resource_federation_path, res_id=new_res.short_id,
-                                            file_name=os.path.basename(f.fed_resource_file.name))))
+            ori_file_path = f.fed_resource_file.name
+            idx1 = ori_file_path.find(settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+            # resource uuid is 32 bits, find idx2 to start right after resource id
+            idx2 = idx1 + len(settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE) + 1 + 32
+            if idx1 > 0:
+                new_file_path = ori_file_path[0:idx1] + \
```

Comment:
I thought about it initially, then decided os.path.join() is an overkill and does not apply here since we explicitly find the index and form a correct path without involving ```/``` separator issue on the path. The direct string concatenation is simpler and works better here.